### PR TITLE
Convert modules to Friendly_api

### PR DESCRIPTION
**Pull request by maiste**
This is the main part of the job to convert `Raw` JSON to typed structures. The last commit should remove the `Api` and the `Friendly_api` module as discussed in #68.

- [X] Users
- [x] Auth
- [x] Ips
- [x] Projects
- [x] Devices

### CODE REVIEW

File: src/lib/equinoxe/equinoxe.ml [hunk 1]
```diff
@@ -576,4 +423,283 @@ struct
...
+      let path = Format.sprintf "devices/%s/events" id in
+      let* json = Http.get ~t ~path in
+      try access "events" json |> Ezjsonm.get_list Event.t_of_json |> return
+      with Ezjsonm.Parse_error (v, err) ->
+        fail_with_parsing ~name:"Device.get_events_from" ~err ~json v
+
+    let execute_action_on t ~id ~action =
+      match action with
+      | Unknown_action s ->
+          let msg = Format.sprintf "Can't execute the following action: %s" s in
```
[the remark this answers is not in the document]
Comment (maiste) [reply src/lib/equinoxe/equinoxe.ml:687]: It's WIP :wink: I will raise an exception if it's not found.

File: src/lib/equinoxe/equinoxe.ml [hunk 2]
```diff
@@ -149,209 +564,174 @@ module Make (B : Backend) : API with type 'a io = 'a B.io = struct
...
+    }
 
-    let request ~t ~path http_request =
-      http_request ~headers:(headers ~token:t.token) ~url:(url ~t ~path)
+    type setter = Hostname of string
 
-    let run ~t ~path http_request =
-      let* body = request ~t ~path http_request in
-      get_json body
+    let build ~plan ~os ~location = { hostname = None; plan; os; location }
```
Comment (art-w) [inline src/lib/equinoxe/equinoxe.ml:598]: The `build` could accept an optional `?hostname` :)

File: src/lib/equinoxe/equinoxe.ml [hunk 3]
```diff
@@ -70,36 +100,416 @@ module Make (B : Backend) : API with type 'a io = 'a B.io = struct
...
+          id = access "id" json |> Ezjsonm.get_string;
+          state = access "state" json |> Ezjsonm.get_string |> State.of_string;
+          event_type = access "type" json |> Ezjsonm.get_string;
+          body = access "body" json |> Ezjsonm.get_string;
+          created_at =
+            access "created_at" json
+            |> Ezjsonm.get_string
+            |> Date.Parser.from_iso;
+        }
+      with Failure _ -> raise_wrong_date ~name:"Event.t_of_json" json
```
Comment (art-w) [inline src/lib/equinoxe/equinoxe.ml:480]: Perhaps we could add a wrapper around `Date.Parser.from_iso` to catch its `Failure` uniformly?

**Review by art-w — APPROVED** [review]
Lovely! The documentation has a few misnamed functions due to all the renaming, but otherwise it looks great :)

**Comment by maiste** [comment]
Oops! My mistake. I get through the code to correct them and then I'll merge the PR. Thanks for the review :+1:

**Pull request merged**